runtime/src/lib.rs

```diff
@@ -29,32 +85,57 @@ fn wait_for_cmd() {
     //}
 }
 
-fn handle_command() -> CaliptraResult<()> {
+fn handle_command() -> CaliptraResult<MboxStatusE> {
     let cmd_id = Mailbox::cmd();
+    let dlen = Mailbox::dlen() as usize;
     let dlen_words = Mailbox::dlen_words() as usize;
     let mut buf = [0u32; 1024];
     Mailbox::copy_from_mbox(buf.get_mut(..dlen_words).ok_or(err_u32!(InternalErr))?);
 
-    // TODO: Actually handle command
-    cprintln!("[rt] Received command={}, len={}", cmd_id, Mailbox::dlen());
+    if dlen > buf.len() * 4 {
+        // dlen larger than max message
+        return Err(err_u32!(InsufficientMemory));
+    }
 
-    // Write response
-    let out_buf = [0xFFFFFFFFu32; 4];
-    Mailbox::set_dlen((out_buf.len() * size_of::<u32>()) as u32);
-    Mailbox::copy_to_mbox(&out_buf);
+    // Convert command buffer to byte slice this is safe since we have already
+    // confirmed the buffer is longer than dlen
+    let cmd_bytes: &[u8] =
+        unsafe { ::core::slice::from_raw_parts(buf.as_ptr() as *const u8, dlen) };
```

Comment:
This code is unsound because the lifetime of cmd_bytes is not tied to buf, allowing cmd_bytes to alias modifications to buf. (I can modify buf while the cmd_bytes reference is alive).

Please don't use unsafe code in app code. In this case, you should probably use zerocopy to do this:

```
let cmd_bytes = &buf.as_bytes()[..dlen];
```

Reply:
unsafe code should be placed into its own module, and have extensive test cases that execute under MIR to check for undefined behavior.

Reply:
FWIW, I would probably write this as:

```
    let Some(cmd_bytes) = buf.as_bytes().get(..dlen) else {
        return Err(err_u32!(InsufficientMemory));
    };
```